partiql-ast/src/main/java/org/partiql/ast/sql/SqlDialect.kt

```diff
@@ -757,23 +775,25 @@ public abstract class SqlDialect : AstVisitor<SqlBlock, SqlBlock>() {
             else -> throw UnsupportedOperationException("Cannot print $node")
         }
         t = t concat list("", "") { node.keys }
-        t = if (node.asAlias != null) t concat " GROUP AS ${node.asAlias.sql()}" else t
+        val asAlias = node.asAlias
+        t = if (asAlias != null) t concat " GROUP AS ${asAlias.sql()}" else t
```

Comment:
(self-review): previously, when we used public fields, we could use the smart-casting of these nullable fields to non-null values (e.g. call `node.asAlias.sql()` directly). Since we switch to use `getters`, will need to define a variable first that we perform null-checks on.